pyfesom2/transport.py

```diff
@@ -0,0 +1,1083 @@
+"""
+Module for computing transports across sections from fesom2 output
+Author: Finn Heukamp (finn.heukamp@awi.de)
+Initial version: 23.11.2021
+"""
+
+import warnings
+from os.path import isfile, isdir
+import xarray as xr
+import numpy as np
+import shapely.geometry as sg
+import pyproj
+import pymap3d as pm
+from dask.diagnostics import ProgressBar
+from tqdm.notebook import tqdm
+from .load_mesh_data import load_mesh
+from .ut import vec_rotate_r2g, get_no_cyclic, cut_region
+
+
+def _ProcessInputs(section, mesh_path, data_path, years, mesh_diag_path):
+    '''
+    process_inputs.py
+
+    Processes inputs and aborts if inputs are wrong
+
+    Inputs
+    ------
+    section (list, str)
+        either a list of the form [lon_start, lon_end, lat_start, lat_end] or a string for a preset section
+    mesh_path (str)
+        directory where the mesh files are stored
+    data_path (str)
+        directory where the data is stored
+    years (np.ndarray)
+    mesh_diag_path (str: optional, default=None)
+        directory where the mesh_diag file is stored, if None it is assumed to be located in data_path
+
+    Returns
+    -------
+    mesh (fesom.mesh object)
+        fesom mesh
+    mesh_diag (xr.dataset)
+        fesom mesh diag
+    section (dict)
+        section dictionary containing additional information
+    files (list)
+        list of velocity files
+
+
+        '''
+
+    # Check the input data types
+    if not isinstance(section, list) | isinstance(section, str):
+        raise ValueError(
+            'The section must be a list of form [lon_start, lon_end, lat_start, lat_end] or a string for a preset section ("FS", "BSO", "BSX", ...)')
+
+    if isinstance(section, list) & (len(section) != 4):
+        raise ValueError(
+            'The section must be a list of form [lon_start, lon_end, lat_start, lat_end]')
+
+    if not isinstance(mesh_path, str):
+        raise ValueError('mesh path must be a string')
+
+    if (mesh_diag_path != None) & (not isinstance(mesh_diag_path, str)):
+        raise ValueError('mesh diag path must be a string')
+
+    if not isdir(mesh_path):
+        raise FileExistsError('The mesh folder does not exist!')
+
+    if mesh_diag_path == None:
+        mesh_diag_path = mesh_path + 'fesom.mesh.diag.nc'
+        if not isfile(mesh_diag_path):
+            raise FileExistsError(
+                'The fesom.mesh.diag.nc file is not located in data_path! Please specify the absolute path!')
+
+    elif isinstance(mesh_diag_path, str):
+        if not isfile(mesh_diag_path):
+            raise FileExistsError('The mesh diag file does not exist!')
+
+    # Load the mesh and the mesh_diag files
+    mesh = load_mesh(mesh_path)
```

Comment:
Each time you use a function, the mesh will be created inside the function. It's fine for small meshes, but for big ones it can become a problem. Would that be more logical to get the `mesh` object as an input. For the moment it's fine to keep it like this, but might cause some slowdowns for larger meshes.